pkg/providers/openstack/delete_image.go

```diff
@@ -2,8 +2,55 @@ package openstack
 
 import (
 	"github.com/emc-advanced-dev/pkg/errors"
+	"github.com/emc-advanced-dev/unik/pkg/types"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/imageservice/v2/images"
 )
 
 func (p *OpenstackProvider) DeleteImage(id string, force bool) error {
-	return errors.New("not yet supportded for openstack", nil)
+	image, err := p.GetImage(id)
+	if err != nil {
+		return errors.New("retrieving image", err)
+	}
+
+	// Delete instances of this image.
+	instances, err := p.ListInstances()
+	if err != nil {
+		return errors.New("retrieving list of instances", err)
+	}
+	for _, instance := range instances {
+		if instance.ImageId == image.Id {
+			if !force {
+				return errors.New("instance "+instance.Id+" found which uses image "+image.Id+"; try again with force=true", nil)
+			} else {
+				err = p.DeleteInstance(instance.Id, true)
+				if err != nil {
+					return errors.New("failed to delete instance "+instance.Id+" which is using image "+image.Id, err)
+				}
+			}
+		}
+	}
+
+	clientGlance, err := p.newClientGlance()
+	if err != nil {
+		return err
+	}
+
+	if err := deleteImage(clientGlance, image.Id); err != nil {
+		return errors.New("failed deleting image "+image.Id, err)
+	}
+
+	// Update state.
+	if err := p.state.ModifyImages(func(imageList map[string]*types.Image) error {
+		delete(imageList, image.Id)
+		return nil
```

Comment:
@ilackarms I have some problems understanding how `p.state.ModifyImages()` works. I can see function signature [here](https://github.com/emc-advanced-dev/unik/blob/master/pkg/state/state.go#L11) but just can't figure out how on Earth is this working - can't find a line where you actually save state to the yaml. I can, on the other hand, confirm that it works :smile: